saleor/plugins/openid_connect/plugin.py

```diff
@@ -0,0 +1,254 @@
+            ),
+            "label": "Client Secret",
+        },
+        "enable_refresh_token": {
+            "type": ConfigurationTypeField.BOOLEAN,
+            "help_text": (
+                "Determine if the refresh token should be also fetched from provider. "
+                "By disabling it, users will need to re-login after the access token "
+                "expired. By enabling it, frontend apps will be able to refresh the "
+                "access token."
+            ),
+            "label": "Enable refreshing token",
+        },
+        "oauth_authorization_url": {
+            "type": ConfigurationTypeField.STRING,
+            "help_text": ("The endpoint used to redirect user to authorization page"),
+            "label": "OAuth Authorization URL",
+        },
+        "oauth_token_url": {
+            "type": ConfigurationTypeField.STRING,
+            "help_text": (
+                "The endpoint to exchange an Authorization Code for a Token."
+            ),
+            "label": "OAuth Token URL",
+        },
+        "json_web_key_set_url": {
+            "type": ConfigurationTypeField.STRING,
+            "help_text": (
+                "The JSON Web Key Set (JWKS) is a set of keys containing the public "
+                "keys used to verify any JSON Web Token (JWT) issued by the "
+                "authorization server and signed using the RS256 signing algorithm."
+            ),
+            "label": "JSON Web Key Set URL",
+        },
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Convert to dict to easier take config elements
+        configuration = {item["name"]: item["value"] for item in self.configuration}
+        self.config = OpenIDConnectConfig(
+            client_id=configuration["client_id"],
+            client_secret=configuration["client_secret"],
+            enable_refresh_token=configuration["enable_refresh_token"],
+            json_web_key_set_url=configuration["json_web_key_set_url"],
+            authorization_url=configuration["oauth_authorization_url"],
+            token_url=configuration["oauth_token_url"],
+        )
+        self.oauth = self._get_oauth_session()
+
+    @classmethod
+    def validate_plugin_configuration(cls, plugin_configuration: "PluginConfiguration"):
+        """Validate if provided configuration is correct."""
+        configuration = plugin_configuration.configuration
+        configuration = {item["name"]: item["value"] for item in configuration}
+        if plugin_configuration.active:
+            incorrect_fields = []
+            urls_to_validate = {
+                "json_web_key_set_url": configuration["json_web_key_set_url"],
+                "oauth_authorization_url": configuration["oauth_authorization_url"],
+                "oauth_token_url": configuration["oauth_token_url"],
+            }
+            incorrect_fields.extend(get_incorrect_or_missing_urls(urls_to_validate))
```

Comment:
Not needed to define it earlier
```suggestion
            incorrect_fields = get_incorrect_or_missing_urls(urls_to_validate)
```